packages/daemon/src/types.d.ts

```diff
@@ -188,6 +241,15 @@ export type ProvideValueForFormulaIdentifier = (
 export type ProvideControllerForFormulaIdentifier = (
   formulaIdentifier: string,
 ) => Controller;
+export type ProvideControllerForFormulaIdentifierAndResolveHandle = (
+  formulaIdentifier: string,
+) => Promise<Controller>;
+
+export interface Handle {}
+export interface InternalHandle {
+  targetFormulaIdentifier: string;
+}
```

Comment:
We should document what handles are somewhere, and docstrings on these interfaces is as good a place as any. The distinction between `Handle` and `InternalHandle` and when to use one vs. the other would also be good.